File: app/types/dispatch.ts
export type DispatchLoadStatus =
  | "Scheduled"
  | "Assigned"
  | "Loading"
  | "En Route"
  | "On Site"
  | "Pouring"
  | "Completed"
  | "Delayed"
  | "Cancelled"
  | string;

export type DispatchLoad = {
  id: string;
  company_id?: string;

  order_number: string;
  customer_name: string;
  job_number: string;

  plant?: string;
  truck_number: string;
  driver_name?: string;

  mix: string;
  quantity: number;

  status: DispatchLoadStatus;
  scheduled_time: string;

  ticket_token?: string;
  ticket_number?: string;

  created_at?: string;
  updated_at?: string;
};

export type DispatchStats = {
  scheduled_loads: number;
  dispatched: number;
  on_site: number;
  needs_attention: number;
};

File: backend/app/schemas/dispatch.py
from pydantic import BaseModel


class DispatchStats(BaseModel):
    scheduled_loads: int
    dispatched: int
    on_site: int
    needs_attention: int


class DispatchLoad(BaseModel):
    id: str
    company_id: str
    order_number: str
    customer_name: str
    job_number: str
    plant: str
    truck_number: str
    driver_name: str
    mix: str
    quantity: float
    status: str
    scheduled_time: str
    ticket_token: str
    ticket_number: str
    created_at: str
    updated_at: str